<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainUI</class>
 <widget class="QWidget" name="MainUI">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1219</width>
    <height>557</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>基于粒计算的多模态数据多粒度融合仿真系统</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QTextEdit" name="textView_1">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>30</y>
     <width>240</width>
     <height>51</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">border:2px solid rgb(186,186,186);
border-radius:10px
</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>300</x>
     <y>10</y>
     <width>91</width>
     <height>87</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>微软雅黑</family>
     <pointsize>18</pointsize>
     <weight>75</weight>
     <italic>false</italic>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">#label_2{
	border-radius:4px;
	background-color: transparent;
	font: 18pt &quot;微软雅黑&quot;;
	font-weight:bold;
	color: rgb(0,0,0);
}

</string>
   </property>
   <property name="text">
    <string>数据集:</string>
   </property>
  </widget>
  <widget class="QTabWidget" name="tabWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>110</y>
     <width>971</width>
     <height>451</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>18</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">
QTabWidget::tab-bar { 
	alignment: center; 
} 
QTabBar::tab:hover{ 
	color: rgb(0, 0, 0);
	background-color: transparent;
}
QTabWidget::pane{ 
	border:none;
}
QTabBar::tab {
    background-color: transparent; /* 标签页背景颜色 */
    font-family: 微软雅黑; /* 标签页字体类型 */
    font-size: 24px; /* 标签页字体大小 */
    color: rgb(255, 255, 255); /* 标签页字体颜色 */
    padding: 10px; /* 内边距 */
    border: 1px solid rgb(150, 150, 150); /* 边框 */
    border-top-left-radius: 5px; /* 左上角圆角 */
    border-top-right-radius: 5px; /* 右上角圆角 */
}
QTabBar::tab:selected {
    background-color: transparent; /* 选中标签页的背景颜色 */
    color: rgb(0, 0, 0); /* 选中标签页的字体颜色 */
}
*{
	background-color: transparent;
}


</string>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="tab">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <attribute name="title">
     <string>粒结构建立</string>
    </attribute>
    <widget class="QTextEdit" name="textShow_1">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>10</y>
       <width>891</width>
       <height>371</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>微软雅黑</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">
background: transparent;  
color: rgb(0, 0, 0);  
</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>粒结构优化</string>
    </attribute>
    <widget class="QTextEdit" name="textShow_2">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>10</y>
       <width>901</width>
       <height>371</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>微软雅黑</family>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">
background: transparent;  
color: rgb(0, 0, 0);  
</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>问题求解</string>
    </attribute>
    <widget class="QTextEdit" name="textShow_3">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>10</y>
       <width>901</width>
       <height>371</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>微软雅黑</family>
       <pointsize>12</pointsize>
      </font>
     </property>
    </widget>
    <widget class="QFrame" name="frame_2">
     <property name="geometry">
      <rect>
       <x>-41</x>
       <y>-160</y>
       <width>991</width>
       <height>561</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">
background: transparent;  
color: rgb(0, 0, 0);  
</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
    </widget>
    <zorder>frame_2</zorder>
    <zorder>textShow_3</zorder>
   </widget>
  </widget>
  <widget class="QFrame" name="frame">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>231</width>
     <height>561</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">
background-image: url(:/images/bg2_cs01.jpeg);
  background-position: center;</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="ahu01">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>440</y>
      <width>111</width>
      <height>111</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">    background: transparent;  
    color: white;  </string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="pixmap">
     <pixmap resource="images/image.qrc">:/images/ahu_02.jpg</pixmap>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="gridFrame">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>221</width>
     <height>461</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="5" column="0">
     <widget class="QPushButton" name="help">
      <property name="font">
       <font>
        <family>黑体</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">#help{
	background: transparent;  
    color: rgb(0, 0, 0);  
	text-align: left;
	border-radius:10px
}
#help:hover{
	icon:url(:/images/help_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#help:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
      </property>
      <property name="text">
       <string>帮助文档</string>
      </property>
      <property name="icon">
       <iconset resource="images/image.qrc">
        <normaloff>:/images/help.svg</normaloff>:/images/help.svg</iconset>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QPushButton" name="buildGranular">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="font">
       <font>
        <family>黑体</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">#buildGranular{
text-align: left;
	
	background-color: transparent;
	color: rgb(0, 0, 0);
	border-radius:10px
}
#buildGranular:hover{
	icon:url(:/images/newlybuild_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#buildGranular:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
      </property>
      <property name="text">
       <string>粒结构建立</string>
      </property>
      <property name="icon">
       <iconset resource="images/image.qrc">
        <normaloff>:/images/newlybuild.svg</normaloff>:/images/newlybuild.svg</iconset>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QPushButton" name="optGranular">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="font">
       <font>
        <family>黑体</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">#optGranular{
text-align: left;
	
	background: transparent;  
    color: rgb(0, 0, 0);  
	border-radius:10px
}
#optGranular:hover{
	icon:url(:/images/smart-optimization_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#optGranular:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
      </property>
      <property name="text">
       <string>粒结构优化</string>
      </property>
      <property name="icon">
       <iconset resource="images/image.qrc">
        <normaloff>:/images/smart-optimization.svg</normaloff>:/images/smart-optimization.svg</iconset>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QPushButton" name="pushProblems">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="font">
       <font>
        <family>黑体</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">#pushProblems{
    background: transparent;  
    color: white;  
	text-align: left;
	color: rgb(0, 0, 0);
	border-radius:10px
}
#pushProblems:hover{
	icon:url(:/images/help_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#pushProblems:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
      </property>
      <property name="text">
       <string>问题求解</string>
      </property>
      <property name="icon">
       <iconset resource="images/image.qrc">
        <normaloff>:/images/tips.svg</normaloff>:/images/tips.svg</iconset>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QPushButton" name="pushView_1">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="font">
       <font>
        <family>黑体</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">#pushView_1{
	text-align: left;
	background-color: transparent;
	color: rgb(0, 0, 0);
	border-radius:10px
}
#pushView_1:hover{
	icon:url(:/images/inbox-out_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#pushView_1:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
      </property>
      <property name="text">
       <string>导入数据集</string>
      </property>
      <property name="icon">
       <iconset resource="images/image.qrc">
        <normaloff>:/images/inbox-out.svg</normaloff>
        <normalon>:/images/inbox-out_hover.svg</normalon>
        <activeon>:/images/inbox-out_hover.svg</activeon>
        <selectedoff>:/images/inbox-out.svg</selectedoff>
        <selectedon>:/images/inbox-out_hover.svg</selectedon>:/images/inbox-out.svg</iconset>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="btnConfig">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="font">
       <font>
        <family>黑体</family>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">#btnConfig{
text-align: left;
	
	background-color: transparent;
	color: rgb(0, 0, 0);
	border-radius:10px
}
#btnConfig:hover{
	icon:url(:/images/file-settings-one_hover.svg);
	color: rgb(144, 19, 158);
	background-color:transparent;
}
#btnConfig:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
      </property>
      <property name="text">
       <string>配置</string>
      </property>
      <property name="icon">
       <iconset resource="images/image.qrc">
        <normaloff>:/images/file-settings-one.svg</normaloff>:/images/file-settings-one.svg</iconset>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QFrame" name="bg">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>0</y>
     <width>991</width>
     <height>561</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-image: url(:/images/bg_cs01.jpg);
background-position: center;</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QComboBox" name="selectAlgorithm">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>60</y>
      <width>251</width>
      <height>32</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>黑体</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">#selectAlgorithm{
	border-radius:11px;
	background-color: rgb(184, 184, 184);
	color: rgb(56,56,56);
	background: transparent;  
}
#selectAlgorithm:hover{
	border-radius:0px;
	background-color: rgb(184, 184, 184);
	border-top-left-radius:10px;
	border-top-right-radius:10px;
	border-bottom-right-radius:10px;
	color: rgb(56,56,56);
	background: transparent;  
}
#selectAlgorithm::drop-down{
	border-top-right-radius:5px;
	border-bottom-right-radius:5px;
	background-color: rgb(40, 40, 40);
	color: rgb(40,40,40);
	min-width:30px;
	background: transparent;  
}
#selectAlgorithm::down-arrow{
	image: url(:/images/rows.png);
	height:28px;
	width:28px;
}
#selectAlgorithm QComboBox{
	font-size:20px;
	font-weight:bold;
background: transparent;  
}
#selectAlgorithm QComboBox::item{
	color: rgb(40, 40, 40);
	height:30px;
    selection-color: rgb(186,186,186);
    background-color:rgb(80, 80, 80);
}

</string>
    </property>
    <property name="placeholderText">
     <string/>
    </property>
    <item>
     <property name="text">
      <string>满意粒度层选择</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>最优粒度层选择</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="selectAlgorithm_2">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>10</y>
      <width>251</width>
      <height>32</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>黑体</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true"> 
#selectAlgorithm_2{
	border-radius:11px;
	background-color: rgb(184, 184, 184);
	color: rgb(56,56,56);
background: transparent;  
}
#selectAlgorithm_2:hover{
	border-radius:0px;
	background-color: rgb(184, 184, 184);
	border-top-left-radius:10px;
	border-top-right-radius:10px;
	border-bottom-right-radius:10px;
	color: rgb(56,56,56);
	background: transparent;  
}
#selectAlgorithm_2::drop-down{
	border-top-right-radius:5px;
	border-bottom-right-radius:5px;
	background-color: rgb(40, 40, 40);
	color: rgb(40,40,40);
	min-width:30px;
	background: transparent;  
}
#selectAlgorithm_2::down-arrow{
	image: url(images/rows.png);
	height:28px;
	width:28px;
}
#selectAlgorithm_2 QComboBox{
	font-size:20px;
	font-weight:bold;
}
#selectAlgorithm_2 QComboBox::item{
	color: rgb(40, 40, 40);
	height:30px;
    selection-color: rgb(186,186,186);
    background-color:rgb(80, 80, 80);
	background: transparent;  
}


</string>
    </property>
    <property name="currentText">
     <string>DecisionTree</string>
    </property>
    <property name="placeholderText">
     <string/>
    </property>
    <item>
     <property name="text">
      <string>DecisionTree</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>KNN</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>SVC</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>AdaBoost</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="resetView_1">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>30</y>
      <width>121</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>黑体</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">#resetView_1{
	background-color:transparent;
	color: rgb(0, 0, 0);
	border-radius:10px
}
#resetView_1:hover{
	icon:url(images/refresh_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#resetView_1:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
    </property>
    <property name="text">
     <string>重置</string>
    </property>
    <property name="icon">
     <iconset resource="images/image.qrc">
      <normaloff>:/images/refresh.svg</normaloff>:/images/refresh.svg</iconset>
    </property>
   </widget>
   <widget class="QPushButton" name="tip3">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>40</y>
      <width>21</width>
      <height>28</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;导入的文件必须是.csv格式的文件&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true">#tip3{
    background: transparent;  
    color: white;  
	text-align: left;
	color: rgb(0, 0, 0);
	border-radius:10px
}
#tip3:hover{
	icon:url(:/images/help_hover.svg);
	color: rgb(144, 19, 158);
	background-color: transparent;
}
#tip3:pressed{
	padding-top:5px;
	padding-left:5px
}
</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="images/image.qrc">
      <normaloff>:/images/help.svg</normaloff>:/images/help.svg</iconset>
    </property>
   </widget>
   <widget class="QFrame" name="verticalFrame">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>0</y>
      <width>141</width>
      <height>101</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
background: transparent;  
color: rgb(0, 0, 0);  
</string>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QRadioButton" name="r1">
       <property name="font">
        <font>
         <family>微软雅黑 Light</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>属性拼接</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="r2">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑 Light</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>结果融合</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="radioButton">
       <property name="font">
        <font>
         <family>微软雅黑 Light</family>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>特征融合</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <zorder>verticalFrame</zorder>
   <zorder>selectAlgorithm</zorder>
   <zorder>selectAlgorithm_2</zorder>
   <zorder>resetView_1</zorder>
   <zorder>tip3</zorder>
  </widget>
  <zorder>bg</zorder>
  <zorder>frame</zorder>
  <zorder>gridFrame</zorder>
  <zorder>textView_1</zorder>
  <zorder>label_2</zorder>
  <zorder>tabWidget</zorder>
 </widget>
 <resources>
  <include location="images/image.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>resetView_1</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>reSetView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1081</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>848</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushView_1</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>importView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>108</x>
     <y>-28</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnConfig</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>configGranular()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>-123</x>
     <y>111</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buildGranular</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>buildGranularW()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>151</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>-61</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>optGranular</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>optGranularW()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>-3</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushProblems</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>resolveProblemsW()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>182</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>-37</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>help</sender>
   <signal>clicked()</signal>
   <receiver>MainUI</receiver>
   <slot>helps()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>407</y>
    </hint>
    <hint type="destinationlabel">
     <x>-137</x>
     <y>407</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>importView()</slot>
  <slot>reSetView()</slot>
  <slot>buildGranularW()</slot>
  <slot>optGranularW()</slot>
  <slot>resolveProblemsW()</slot>
  <slot>configGranular()</slot>
  <slot>helps()</slot>
  <slot>selectRadio()</slot>
 </slots>
</ui>
